### PR TITLE
Add support for multiple delimiters and the `Delimiter` keyword

### PR DESCRIPTION
**Pull request by silverbullettruck2001**
Modified the ansi dialect to allow for multiple delimiters for languages that want to override this to add multiple delimiters.  Also, updated the mysql dialect to allow parsing of the `Delimiter` keyword.

### CODE REVIEW

**Comment by silverbullettruck2001** [comment]
@barrywhart here's the PR with the updates that don't seem to be identifying the trailing `DELIMITER ;` and the ending `~` as expected.  Thoughts?

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 1]
```diff
@@ -181,6 +178,7 @@
 
 ansi_dialect.add(
     # Real segments
+    DelimiterSegment=OneOf(Ref("SemicolonSegment")),
```
Comment (NiallRees) [inline src/sqlfluff/dialects/dialect_ansi.py:181]:
```suggestion
    DelimiterSegment=Ref("SemicolonSegment"),
```

File: src/sqlfluff/dialects/dialect_ansi.py [hunk 2]
```diff
@@ -1,11 +1,8 @@
 """The core ANSI dialect.
-
```
Comment (NiallRees) [inline src/sqlfluff/dialects/dialect_ansi.py:2]: Please could you add all the line breaks back that seem to have gone?

**Review by NiallRees — CHANGES_REQUESTED** [review]
Thanks for this @silverbullettruck2001 !

**Review by silverbullettruck2001 — COMMENTED** [review]
This is done

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 3]
```diff
@@ -90,3 +93,39 @@ class CreateTableStatementSegment(
...
+
+
+@mysql_dialect.segment()
+class DelimiterStatement(BaseSegment):
+    """DELIMITER statement."""
+
+    type = "delimiter_statement"
+    match_grammar = Sequence(
+        "DELIMITER",
+    )
```
Comment (NiallRees) [inline src/sqlfluff/dialects/dialect_mysql.py:122]:
```suggestion
    match_grammar = Ref.keyword("DELIMITER")
```

File: src/sqlfluff/dialects/dialect_mysql.py [hunk 4]
```diff
@@ -62,7 +65,7 @@
         CodeSegment,
         name="quoted_literal",
         type="literal",
-        trim_chars=('"',),
+        trim_chars=('"'),
```
Comment (barrywhart) [inline src/sqlfluff/dialects/dialect_mysql.py:68]: It looks like here, you changed a tuple that contains a string to just be a string. Was this change intentional? If not, can you change it back?

**Review by barrywhart — CHANGES_REQUESTED** [review]
Looks good overall. I noticed one change that looks like it may have been accidental. Can you check and maybe undo that change?

File: src/sqlfluff/dialects/dialect_mysql.py (hunk 4 above)
Comment (silverbullettruck2001) [reply src/sqlfluff/dialects/dialect_mysql.py]: This has been fixed now

**Review by barrywhart — APPROVED** [review]

**Review by NiallRees — APPROVED** [review]
